integration/docker/Dockerfile

```diff
@@ -32,10 +32,37 @@ RUN if [ ${ENABLE_DYNAMIC_USER} = "true" ] ; then \
        chmod -R 777 /opt/* ; \
     fi
 
-# instead of ubuntu:16.04, use alluxio/alluxio-ubuntu:ubuntu1604-customize
-# applies optimizations in libfuse on top of ubuntu:16.04
-# it is based off cheyang/fuse2:ubuntu1604-customize
-FROM alluxio/alluxio-ubuntu:1604-customize
+# The following setup is modified from cheyang/fuse2:ubuntu1604-customize to be applied on alpine
+WORKDIR /
+
+# fuse3 requires meson 0.42, available from stretch-backports
+# wget depends on ca-certificates
+
+RUN \
+  apk update && app upgrade && \
+  apk add --no-cache build-base ca-certificates pkgconfig wget eudev git && \
+  apk add --no-cache gcc make cmake gettext-dev libtool autoconf automake
+
+RUN \
+   git clone https://github.com/cheyang/libfuse.git && \
+   cd libfuse && \
+   git checkout fuse_2_9_5_customize_multi_threads_no_logging
+
+#ADD libfuse /libfuse
+
+WORKDIR "/libfuse"
+
+RUN bash makeconf.sh && \
+  ./configure && \
+  make -j8 && \
+  make install
```

Comment:
after libfuse is created, can we remove all libfuse related src code and compilation tools to further reduce the image size?

Reply:
We could, but the installation and uninstallation have to be on the same layer, which means we will have a giant RUN command, i.e. install all packages, install libfuse, uninstall all packages, all under the same RUN. Do we want to do that?